internal/kafka/internal/clusters/ocm_provider.go

```diff
@@ -534,3 +534,22 @@ func (o *OCMProvider) GetClusterResourceQuotaCosts() ([]types.QuotaCost, error)
 	}
 	return quotaCostList, nil
 }
+
+func (o *OCMProvider) CheckIfOrganizationIsTheClusterOwner(externalOrganizationID, clusterID, clusterExternalID string) error {
+	organizationID, err := o.ocmClient.GetOrganisationIdFromExternalId(externalOrganizationID)
+	if err != nil || shared.StringEmpty(organizationID) {
+		return svcErrors.NewWithCause(svcErrors.ErrorGeneral, err, "failed to retrieve organization details")
+	}
+
+	search := fmt.Sprintf("cluster_id = '%s' AND external_cluster_id = '%s' AND organization_id = '%s'", clusterID, clusterExternalID, organizationID)
+	subcriptions, err := o.ocmClient.GetSubscriptions(0, 1, search)
+	if err != nil {
+		return svcErrors.NewWithCause(svcErrors.ErrorGeneral, err, "failed to validate if the organization with id %q owns the cluster with id %q", externalOrganizationID, clusterID)
+	}
+
+	if len(subcriptions) == 0 {
```

Comment:
Side comment: I wonder if adding an `arrays.IsEmpty(ary []T) bool` function to the array utilities would make these tests more readable.

Reply:
+1 - I'd agree. Ever since there is `shared.StringEmpty` I've been using it almost everywhere.